### ISSUE TITLE
Automation not producing anything after being launched. 

### ISSUE THREAD

**Issue by Theresa2020** [Public-Health-Scotland/posit-workbench-job-scheduler#16]
I tried to run an automation immediately last Wednesday and tried setting one up for Thursday morning at 07:15am to do a parallel run to our cronR job from the old server and it failed with error code 2, didn't know if it was because I shut down my laptop? 

I then tried to do an automation on the Thursday morning and nothing produced (I left the session running with no other analysis to see if that would have an impact), it was set off about 08:30am. Is there anyway we can check what happened to that automation? Below is the one I launched on Thursday morning. 

 
✔ A Workbench Job with id 'S3ViZXJuZXRlczptaS10aW1lLXNlcmllcy13ZDR3Yg==' and name 'MI Time Series' has been launched on the cluster
  'Kubernetes'.

**Comment by ciarag01** [comment]
@jakeybob @rmccreath This is a similar issue to @stephcphs I think

**Comment by jakeybob** [comment]
Closing the laptop shouldn't have any effect. 

If you're still logged in to Posit Workbench then you should be able to take a look at completed jobs (even those that failed) by going to the homepage, and clicking on "completed" Workbench Jobs at the top right, then clicking on the specific job. It should then show any output from the job (i.e. anything the script prints to the console as it's running) as well as show the exit code. Sounds like this job will have status "failed" and exit code 2 though.

![snap](https://user-images.githubusercontent.com/304817/227921488-f2e55da5-3208-4fd4-95c9-ecdabf3e70a0.png)

**Comment by Theresa2020** [comment]
There was this option the other day to look at it (this was where I got the exit 2 code), but no longer there and I didn't close the session. I will re-run another automation this afternoon and let you know if I am still receiving the same issues. Is there anything that we can do to see the status of an automation when we have the workbench id?

**Comment by jakeybob** [comment]
Yep, there's a couple of ways. If you're still logged into the session you launched the job from you can look at the "Workbench Jobs" tab (typically located next to the Console and Terminal tabs at the lower left) and this will list your recent jobs (however this is much the same as the info available from the homepage).

There's also the `rstudioapi::launcherGetJob(job_id)` function, which will return a list of information on that specific `job_id`  i.e. in this case `S3ViZXJuZXRlczptaS10aW1lLXNlcmllcy13ZDR3Yg==`

It's a long list of stuff, so I wrote a wee function to just print out the more useful things (but also not sure if this will work after logging out and back in again).

```
job_id <- launch_workbench_job(job_name = "Test Workbench Job",
                               project_path = here::here(),
                               script = "job_tests.R",
                               n_cpu = 0.25,
                               n_ram = 128)

# print out all job info
rstudioapi::launcherGetJob(job_id)

# quick function to print out useful job info
show_job_info <- function(job_id){
  job_info <- rstudioapi::launcherGetJob(job_id)
  
  c(user = job_info$user, 
    job_name = job_info$name,
    submitted = format(as.POSIXct(job_info$submissionTime, format = "%Y-%m-%dT%H:%M:%S"), "%Y%m%d_%H:%M:%S"),
    last_updated = format(as.POSIXct(job_info$lastUpdateTime, format = "%Y-%m-%dT%H:%M:%S"), "%Y%m%d_%H:%M:%S"),
    status = job_info$status,
    exit_code = job_info$exitCode)
  
}

show_job_info(job_id)
```

**Comment by Theresa2020** [comment]
Thanks Bob, I created an automation today and using the code below, I get an exit code 2, do we know what this is exactly? See output below:
show_job_info(time_series_id)
               user            job_name           submitted        last_updated              status           exit_code
         "therer01"    "MI Time Series" "20230327_11:53:23" "20230327_11:53:30"          "Finished"                 "2"


I was thinking why my other session would have closed over the weekend and it will be due to the sessions being forced closed after 3 days. Terry mentioned that it is being looked into stopping the forced closing of any sessions with automations set up.

Any help on how to get this fixed, is greatly appreciated as this is going to have a big impact on us next week for covid weekly reporting and having to manually run very large extracts and analysts having to start these during working hours, whereas, we have previously had these starting at 07:15am, to have the majority ready for analysts starting.

Thanks,
Theresa
